### ISSUE TITLE
a11y: missing "aria-current" on active link in search modal

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#13820]
**Setup**
- Rancher version:
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**

The current state is not programmatically defined for the currently active page link resulting in screen reader users will be unaware of the currently active state of the link.

Ensure that the current state of the active link is defined using the "aria-current" attribute with an appropriate token value.
Apply the following changes:
• Add the aria-current="page" attribute to <a> element of the link.
• Ensure that the "aria-current" attribute is updated as per user interaction using scripting.

**To Reproduce**

**Result**

**Expected Result**

The screen reader should announce the current page state for the link.

**Screenshots**

**Additional context**

**Comment by yonasberhe23** [comment]
Manual validation required

**Comment by a-arias** [comment]
**TESTS** 
Rancher : v2.12-733fda1f8d80127b7afb834c938cb754dc8e1894-head
UI: master https://github.com/rancher/dashboard/commit/ac02757be26ecca167a9dfb26354c1defcc0d7a7
Fix verified ✅ 

<img width="1584" alt="Image" src="https://github.com/user-attachments/assets/cf997162-6493-41fd-ba20-2985ffa6a66d" />

https://github.com/user-attachments/assets/ded1042a-ce86-41e5-9e09-820b189ce292


<img width="1584" alt="Image" src="https://github.com/user-attachments/assets/be8118de-984e-4a46-86fd-9166b5d5c2c9" />
